version: '3'
services:

  db: # Подключение к базе данных
    image: postgres # Название образа базы данных
#    env_file: # Ссылка на закрытый источник к данным для подключения БД
#      - .env
#    environment: # Ссылка на источник к данным для подключения БД
    environment:
        - POSTGRES_PASSWORD=123456
        - PGDATA=/var/lib/postgresql/data/pgdata
    volumes:
      - pg_data:/var/lib/postgresql/data/pgdata
    ports: # Порт для БД
      - '5432:5432'
    healthcheck: # Проверка подключения БД
      test: [ "CMD-SHELL", "pg_isready -d postgres -U postgres" ]
      interval: 10s # Интервал проверки
      timeout: 5s # Пауза между проверками
      retries: 5 # Количество попыток

  app:
    build: .
    tty: true
    command: sh -c "python manage.py migrate && python manage.py runserver 0.0.0.0:8000"
    volumes:
      - .:/app
    ports:
      - "8000:8000"
    depends_on:
      db:
        condition: service_healthy

  redis:
    image: redis
    ports:
      - '6379:6379'

  celery:
    build: .
    tty: true
    command: celery -A config worker -l INFO
    depends_on:
      - redis
      - app

  celery_beat:
    build: .
    tty: true
    command: celery -A config beat -l info -S django
    depends_on:
      - redis
      - app

volumes:
  pg_data:


